Userland/Utilities/gzip.cpp

```diff
@@ -37,26 +48,32 @@ int main(int argc, char** argv)
         }
         auto file = file_or_error.value();
 
-        auto compressed_file = Compress::GzipCompressor::compress_all(file->bytes());
-        if (!compressed_file.has_value()) {
-            warnln("Failed gzip compressing input file");
+        AK::Optional<ByteBuffer> output_bytes;
+        if (decompress) {
+            output_bytes = Compress::GzipDecompressor::decompress_all(file->bytes());
+        } else {
+            output_bytes = Compress::GzipCompressor::compress_all(file->bytes());
+        }
+
+        if (!output_bytes.has_value()) {
+            warnln("Failed gzip {} input file", decompress ? "decompressing"sv : "compressing"sv);
```

Comment:
Not sure i love this pattern, but :shrug:

Reply:
yeah okay, you'd prefer plain `if/else`?

Reply:
Personally yes, but they're equivalent, so it's not a blocker to me